### ISSUE TITLE
disk device on /run not mounted

### ISSUE THREAD

**Issue by dionysius** [lxc/lxd#5796]
# Required information

 * Distribution: Devuan
 * Distribution version: ascii
 * The output of "lxc info":
```
config: {}
api_extensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_stop_priority
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
- storage
- file_delete
- file_append
- network_dhcp_expiry
- storage_lvm_vg_rename
- storage_lvm_thinpool_rename
- network_vlan
- image_create_aliases
- container_stateless_copy
- container_only_migration
- storage_zfs_clone_copy
- unix_device_rename
- storage_lvm_use_thinpool
- storage_rsync_bwlimit
- network_vxlan_interface
- storage_btrfs_mount_options
- entity_description
- image_force_refresh
- storage_lvm_lv_resizing
- id_map_base
- file_symlinks
- container_push_target
- network_vlan_physical
- storage_images_delete
- container_edit_metadata
- container_snapshot_stateful_migration
- storage_driver_ceph
- storage_ceph_user_name
- resource_limits
- storage_volatile_initial_source
- storage_ceph_force_osd_reuse
- storage_block_filesystem_btrfs
- resources
- kernel_limits
- storage_api_volume_rename
- macaroon_authentication
- network_sriov
- console
- restrict_devlxd
- migration_pre_copy
- infiniband
- maas_network
- devlxd_events
- proxy
- network_dhcp_gateway
- file_get_symlink
- network_leases
- unix_device_hotplug
- storage_api_local_volume_handling
- operation_description
- clustering
- event_lifecycle
- storage_api_remote_volume_handling
- nvidia_runtime
- container_mount_propagation
- container_backup
- devlxd_images
- container_local_cross_pool_handling
- proxy_unix
- proxy_udp
- clustering_join
- proxy_tcp_udp_multi_port_handling
- network_state
- proxy_unix_dac_properties
- container_protection_delete
- unix_priv_drop
- pprof_http
- proxy_haproxy_protocol
- network_hwaddr
- proxy_nat
- network_nat_order
- container_full
- candid_authentication
- backup_compression
- candid_config
- nvidia_runtime_config
- storage_api_volume_snapshots
- storage_unmapped
- projects
- candid_config_key
- network_vxlan_ttl
- container_incremental_copy
- usb_optional_vendorid
- snapshot_scheduling
- container_copy_project
- clustering_server_address
- clustering_image_replication
- container_protection_shift
- snapshot_expiry
- container_backup_override_pool
api_status: stable
api_version: "1.0"
auth: trusted
public: false
auth_methods:
- tls
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
    MI...89
    -----END CERTIFICATE-----
  certificate_fingerprint: be...7a
  driver: lxc
  driver_version: 3.1.0
  kernel: Linux
  kernel_architecture: x86_64
  kernel_version: 4.19.0-0.bpo.4-amd64
  server: lxd
  server_pid: 6279
  server_version: "3.10"
  storage: dir
  storage_version: "1"
  server_clustered: false
  server_name: olaf
  project: default
```

# Issue description

I have a use case where I need to define a disk device mounted on /run before the container is started. I haven't tested all distro's, but so far all modern ones have this issue and I don't know how to solve them.

I have no issues, if the path is somewhere else or I add the device after the container is running - but as I stated, I'd like to define that device before starting (think of a host reboot, that would have that case too).

# Steps to reproduce

```bash
dionysius@olaf:~$ mkdir /tmp/some-dir

dionysius@olaf:~$ lxc init ubuntu:trusty trusty
Creating trusty
dionysius@olaf:~$ lxc config device add trusty mount-test disk source=/tmp/some-dir path=/run/some-dir optional=false
Device mount-test added to trusty
dionysius@olaf:~$ lxc start trusty 
dionysius@olaf:~$ lxc exec trusty -- ls -lah /run
total 24K
drwxr-xr-x  8 root   root    4.0K May 24 11:41 .
drwxr-xr-x 22 root   root    4.0K May 24 11:38 ..
drwxrwxrwt  2 root   root      40 May 24 11:41 lock
drwxr-xr-x  2 root   root    4.0K May 24 11:41 plymouth
drwxrwxrwt  2 root   root      40 May 24 11:41 shm
drwxr-xr-x  2 nobody nogroup 4.0K May 24 11:31 some-dir
drwxr-xr-x  2 root   root    4.0K May 24 11:41 udev
-rw-r--r--  1 root   root       5 May 24 11:41 upstart-udev-bridge.pid
drwxr-xr-x  2 root   root      40 May 24 11:41 user
dionysius@olaf:~$ lxc info trusty --show-log
Name: trusty
Location: none
Remote: unix://
Architecture: x86_64
Created: 2019/05/24 11:38 UTC
Status: Running
Type: persistent
Profiles: default
Pid: 6133
Resources:
  Processes: 6
  CPU usage:
    CPU usage (in seconds): 1
  Memory usage:
    Memory (current): 6.93MB
    Memory (peak): 8.17MB
  Network usage:
    lo:
      Bytes received: 0B
      Bytes sent: 0B
      Packets received: 0
      Packets sent: 0

Log:

lxc trusty 20190524114147.941 WARN     conf - conf.c:lxc_setup_devpts:1641 - Invalid argument - Failed to unmount old devpts instance

dionysius@olaf:~$ lxc init ubuntu:xenial xenial
Creating xenial
dionysius@olaf:~$ lxc config device add xenial mount-test disk source=/tmp/some-dir path=/run/some-dir optional=false  
Device mount-test added to xenial
dionysius@olaf:~$ lxc start xenial
dionysius@olaf:~$ lxc exec xenial -- ls -lah /run
total 4.0K
drwxr-xr-x  3 root root   60 May 24 11:43 .
drwxr-xr-x 22 root root 4.0K May 14 21:23 ..
drwxrwxrwt  2 root root   40 May 24 11:43 lock
# --> some-dir?
dionysius@olaf:~$ lxc info xenial --show-log
Name: xenial
Location: none
Remote: unix://
Architecture: x86_64
Created: 2019/05/24 11:42 UTC
Status: Running
Type: persistent
Profiles: default
Pid: 23523
Resources:
  Processes: 1
  CPU usage:
    CPU usage (in seconds): 0
  Memory usage:
    Memory (current): 1.02MB
    Memory (peak): 1.52MB
  Network usage:
    lo:
      Bytes received: 0B
      Bytes sent: 0B
      Packets received: 0
      Packets sent: 0

Log:

lxc xenial 20190524114311.837 WARN     conf - conf.c:lxc_setup_devpts:1641 - Invalid argument - Failed to unmount old devpts instance

dionysius@olaf:~$ lxc init ubuntu:bionic bionic
Creating bionic
dionysius@olaf:~$ lxc config device add bionic mount-test disk source=/tmp/some-dir path=/run/some-dir optional=false  
Device mount-test added to bionic
dionysius@olaf:~$ lxc start bionic
dionysius@olaf:~$ lxc exec bionic -- ls -lah /run
total 4.0K
drwxr-xr-x  3 root root   60 May 24 11:44 .
drwxr-xr-x 22 root root 4.0K May 14 23:57 ..
drwxrwxrwt  2 root root   40 May 24 11:44 lock
# --> some-dir?
dionysius@olaf:~$ lxc info bionic --show-log
Name: bionic
Location: none
Remote: unix://
Architecture: x86_64
Created: 2019/05/24 11:43 UTC
Status: Running
Type: persistent
Profiles: default
Pid: 3026
Resources:
  Processes: 1
  CPU usage:
    CPU usage (in seconds): 0
  Memory usage:
    Memory (current): 2.56MB
    Memory (peak): 2.96MB
  Network usage:
    lo:
      Bytes received: 0B
      Bytes sent: 0B
      Packets received: 0
      Packets sent: 0

Log:

lxc bionic 20190524114404.669 WARN     conf - conf.c:lxc_setup_devpts:1641 - Invalid argument - Failed to unmount old devpts instance

dionysius@olaf:~$ lxc init images:centos/7 centos7
Creating centos7
dionysius@olaf:~$ lxc config device add centos7 mount-test disk source=/tmp/some-dir path=/run/some-dir optional=false 
Device mount-test added to centos7
dionysius@olaf:~$ lxc start centos7
dionysius@olaf:~$ lxc exec centos7 -- ls -lah /run
total 4.0K
drwxr-xr-x  3 root root   60 May 24 11:54 .
dr-xr-xr-x 18 root root 4.0K May 24 07:17 ..
drwxr-xr-x  4 root root  100 May 24 11:54 systemd
# --> some-dir?
dionysius@olaf:~$ lxc info centos7 --show-log
Name: centos7
Location: none
Remote: unix://
Architecture: x86_64
Created: 2019/05/24 11:54 UTC
Status: Running
Type: persistent
Profiles: default
Pid: 26641
Ips:
  lo:   inet    127.0.0.1
  lo:   inet6   ::1
Resources:
  Processes: 1
  CPU usage:
    CPU usage (in seconds): 0
  Memory usage:
    Memory (current): 1.09MB
    Memory (peak): 1.50MB
  Network usage:
    lo:
      Bytes received: 0B
      Bytes sent: 0B
      Packets received: 0
      Packets sent: 0

Log:

lxc centos7 20190524115447.677 WARN     conf - conf.c:lxc_setup_devpts:1641 - Invalid argument - Failed to unmount old devpts instance
```

# Information to attach

 - [?] Any relevant kernel output (`dmesg`)
```
[10307.732146] audit: type=1400 audit(1558698107.859:40): apparmor="STATUS" operation="profile_load" profile="unconfined" name="lxd-trusty_</var/lib/lxd>" pid=6095 comm="apparmor_parser"
[10391.604608] audit: type=1400 audit(1558698191.731:41): apparmor="STATUS" operation="profile_load" profile="unconfined" name="lxd-xenial_</var/lib/lxd>" pid=23512 comm="apparmor_parser"
[10444.472132] audit: type=1400 audit(1558698244.599:42): apparmor="STATUS" operation="profile_load" profile="unconfined" name="lxd-bionic_</var/lib/lxd>" pid=2997 comm="apparmor_parser"
[11087.448550] audit: type=1400 audit(1558698887.575:43): apparmor="STATUS" operation="profile_load" profile="unconfined" name="lxd-centos7_</var/lib/lxd>" pid=26630 comm="apparmor_parser"
```
 - [x] Container log (`lxc info NAME --show-log`)
see above
 - [x] Container configuration (`lxc config show NAME --expanded`)
nothing special
 - [x] Main daemon log (at /var/log/lxd/lxd.log or /var/snap/lxd/common/lxd/logs/lxd.log)
```
t=2019-05-24T13:38:12+0200 lvl=info msg="Creating container" ephemeral=false name=trusty project=default
t=2019-05-24T13:38:12+0200 lvl=info msg="Created container" ephemeral=false name=trusty project=default
t=2019-05-24T13:41:47+0200 lvl=info msg="Starting container" action=start created=2019-05-24T13:38:12+0200 ephemeral=false name=trusty project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:41:47+0200 lvl=info msg="Started container" action=start created=2019-05-24T13:38:12+0200 ephemeral=false name=trusty project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:42:06+0200 lvl=warn msg="Detected poll(POLLNVAL) event." 
t=2019-05-24T13:42:33+0200 lvl=info msg="Creating container" ephemeral=false name=xenial project=default
t=2019-05-24T13:42:33+0200 lvl=info msg="Created container" ephemeral=false name=xenial project=default
t=2019-05-24T13:43:11+0200 lvl=info msg="Starting container" action=start created=2019-05-24T13:42:33+0200 ephemeral=false name=xenial project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:43:11+0200 lvl=info msg="Started container" action=start created=2019-05-24T13:42:33+0200 ephemeral=false name=xenial project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:43:34+0200 lvl=info msg="Creating container" ephemeral=false name=bionic project=default
t=2019-05-24T13:43:34+0200 lvl=info msg="Created container" ephemeral=false name=bionic project=default
t=2019-05-24T13:44:04+0200 lvl=info msg="Starting container" action=start created=2019-05-24T13:43:34+0200 ephemeral=false name=bionic project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:44:04+0200 lvl=info msg="Started container" action=start created=2019-05-24T13:43:34+0200 ephemeral=false name=bionic project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:44:13+0200 lvl=warn msg="Detected poll(POLLNVAL) event." 
t=2019-05-24T13:54:15+0200 lvl=info msg="Downloading image" alias=centos/7 image=340a3dafb6313b2d94808e7e403e64eee18abf2821f733b94dbc445aa115c3a0 operation=221379b4-0fad-4785-9e62-7a584faa561c server=https://images.linuxcontainers.org trigger=/1.0/operations/221379b4-0fad-4785-9e62-7a584faa561c
t=2019-05-24T13:54:27+0200 lvl=info msg="Image downloaded" alias=centos/7 image=340a3dafb6313b2d94808e7e403e64eee18abf2821f733b94dbc445aa115c3a0 operation=221379b4-0fad-4785-9e62-7a584faa561c server=https://images.linuxcontainers.org trigger=/1.0/operations/221379b4-0fad-4785-9e62-7a584faa561c
t=2019-05-24T13:54:27+0200 lvl=info msg="Creating container" ephemeral=false name=centos7 project=default
t=2019-05-24T13:54:27+0200 lvl=info msg="Created container" ephemeral=false name=centos7 project=default
t=2019-05-24T13:54:47+0200 lvl=info msg="Starting container" action=start created=2019-05-24T13:54:27+0200 ephemeral=false name=centos7 project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:54:47+0200 lvl=info msg="Started container" action=start created=2019-05-24T13:54:27+0200 ephemeral=false name=centos7 project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T13:55:00+0200 lvl=warn msg="Detected poll(POLLNVAL) event." 
```
 - [x] Output of the client with --debug
not client related
 - [x] Output of the daemon with --debug (alternatively output of `lxc monitor` while reproducing the issue)
```
...
t=2019-05-24T14:14:43+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:14:43+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/events?project="
t=2019-05-24T14:14:43+0200 lvl=dbug msg="New event listener: 1650a36e-5133-4d1b-9e80-8bf68f615dab" 
t=2019-05-24T14:14:43+0200 lvl=dbug msg=handling ip=@ method=POST url=/1.0/containers
t=2019-05-24T14:14:43+0200 lvl=dbug msg="Responding to container create" 
t=2019-05-24T14:14:43+0200 lvl=dbug msg="New task operation: 38e24d93-e3a9-4227-b7ba-b281aebda97c" 
t=2019-05-24T14:14:43+0200 lvl=dbug msg="Started task operation: 38e24d93-e3a9-4227-b7ba-b281aebda97c" 
t=2019-05-24T14:14:43+0200 lvl=dbug msg="Connecting to a remote simplestreams server" 
t=2019-05-24T14:14:43+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/operations/38e24d93-e3a9-4227-b7ba-b281aebda97c
t=2019-05-24T14:14:44+0200 lvl=dbug msg="Database error: &errors.errorString{s:\"sql: no rows in result set\"}" 
t=2019-05-24T14:14:44+0200 lvl=dbug msg="Database error: &errors.errorString{s:\"sql: no rows in result set\"}" 
t=2019-05-24T14:14:44+0200 lvl=dbug msg="Image already exists in the db" image=4bfe6258382622a6a67c7930669a831885fae82e7288f170a33330f51f1d757d
t=2019-05-24T14:14:44+0200 lvl=dbug msg="Database error: &errors.errorString{s:\"sql: no rows in result set\"}" 
t=2019-05-24T14:14:44+0200 lvl=info msg="Creating container" ephemeral=false name=trusty project=default
t=2019-05-24T14:14:44+0200 lvl=info msg="Created container" ephemeral=false name=trusty project=default
t=2019-05-24T14:14:44+0200 lvl=dbug msg="Creating DIR storage volume for container \"trusty\" on storage pool \"default\"" 
t=2019-05-24T14:14:45+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:14:45+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:14:49+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:14:49+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:14:53+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:14:53+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:14:56+0200 lvl=dbug msg="Shifting root filesystem \"/var/lib/lxd/containers/trusty/rootfs\" for \"trusty\"" 
t=2019-05-24T14:14:57+0200 lvl=dbug msg="Created DIR storage volume for container \"trusty\" on storage pool \"default\"" 
t=2019-05-24T14:14:57+0200 lvl=dbug msg="Success for task operation: 38e24d93-e3a9-4227-b7ba-b281aebda97c" 
t=2019-05-24T14:14:57+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/containers/trusty
t=2019-05-24T14:14:58+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:14:58+0200 lvl=dbug msg="Disconnected event listener: 1650a36e-5133-4d1b-9e80-8bf68f615dab" 
t=2019-05-24T14:14:58+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:02+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:02+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:06+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:06+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:09+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:09+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/containers/trusty
t=2019-05-24T14:15:09+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/events?project="
t=2019-05-24T14:15:09+0200 lvl=dbug msg="New event listener: de2a99d5-2f0f-432e-b9af-1b8760762691" 
t=2019-05-24T14:15:09+0200 lvl=dbug msg=handling ip=@ method=PUT url=/1.0/containers/trusty
t=2019-05-24T14:15:09+0200 lvl=dbug msg="New task operation: 1cd5334d-75b6-4a5d-8b8e-112028cb89b6" 
t=2019-05-24T14:15:09+0200 lvl=dbug msg="Started task operation: 1cd5334d-75b6-4a5d-8b8e-112028cb89b6" 
t=2019-05-24T14:15:09+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/operations/1cd5334d-75b6-4a5d-8b8e-112028cb89b6
t=2019-05-24T14:15:09+0200 lvl=dbug msg="Success for task operation: 1cd5334d-75b6-4a5d-8b8e-112028cb89b6" 
t=2019-05-24T14:15:10+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:10+0200 lvl=dbug msg="Disconnected event listener: de2a99d5-2f0f-432e-b9af-1b8760762691" 
t=2019-05-24T14:15:10+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:14+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:14+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:16+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:16+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/containers/trusty
t=2019-05-24T14:15:16+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/events?project="
t=2019-05-24T14:15:16+0200 lvl=dbug msg="New event listener: 1157909c-828d-4fa1-9156-2fb2da73c267" 
t=2019-05-24T14:15:16+0200 lvl=dbug msg=handling ip=@ method=PUT url=/1.0/containers/trusty/state
t=2019-05-24T14:15:16+0200 lvl=dbug msg="New task operation: c239cfad-2af0-44b9-816b-73475a6c7ff6" 
t=2019-05-24T14:15:16+0200 lvl=dbug msg="Started task operation: c239cfad-2af0-44b9-816b-73475a6c7ff6" 
t=2019-05-24T14:15:16+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/operations/c239cfad-2af0-44b9-816b-73475a6c7ff6
t=2019-05-24T14:15:16+0200 lvl=info msg="Starting container" action=start created=2019-05-24T14:14:44+0200 ephemeral=false name=trusty project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T14:15:16+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:16+0200 lvl=dbug msg=handling ip=@ method=GET url=/internal/containers/438/onstart
t=2019-05-24T14:15:16+0200 lvl=dbug msg="Scheduler: container trusty started: re-balancing" 
t=2019-05-24T14:15:16+0200 lvl=info msg="Started container" action=start created=2019-05-24T14:14:44+0200 ephemeral=false name=trusty project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T14:15:16+0200 lvl=dbug msg="Success for task operation: c239cfad-2af0-44b9-816b-73475a6c7ff6" 
t=2019-05-24T14:15:19+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:19+0200 lvl=dbug msg="Disconnected event listener: 1157909c-828d-4fa1-9156-2fb2da73c267" 
t=2019-05-24T14:15:19+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:23+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:23+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:27+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:27+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:30+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:30+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=1"
t=2019-05-24T14:15:31+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:31+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
...
t=2019-05-24T14:15:45+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:45+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/events?project="
t=2019-05-24T14:15:45+0200 lvl=dbug msg="New event listener: 0b03ccef-3cb1-464a-ba76-6a6d1b2026a6" 
t=2019-05-24T14:15:45+0200 lvl=dbug msg=handling ip=@ method=POST url=/1.0/containers
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Responding to container create" 
t=2019-05-24T14:15:45+0200 lvl=dbug msg="New task operation: 3edb1db8-639c-42e2-bd6e-0bfaa472b59a" 
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Started task operation: 3edb1db8-639c-42e2-bd6e-0bfaa472b59a" 
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Using SimpleStreams cache entry" expiry=2019-05-24T15:14:44+0200 server=https://cloud-images.ubuntu.com/releases
t=2019-05-24T14:15:45+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/operations/3edb1db8-639c-42e2-bd6e-0bfaa472b59a
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Database error: &errors.errorString{s:\"sql: no rows in result set\"}" 
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Database error: &errors.errorString{s:\"sql: no rows in result set\"}" 
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Image already exists in the db" image=5a01c2c8552f1fafe1d78e863c67e513da76d285afa6f22c4bfa93a802596af9
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Database error: &errors.errorString{s:\"sql: no rows in result set\"}" 
t=2019-05-24T14:15:45+0200 lvl=info msg="Creating container" ephemeral=false name=xenial project=default
t=2019-05-24T14:15:45+0200 lvl=info msg="Created container" ephemeral=false name=xenial project=default
t=2019-05-24T14:15:45+0200 lvl=dbug msg="Creating DIR storage volume for container \"xenial\" on storage pool \"default\"" 
t=2019-05-24T14:15:48+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:48+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:51+0200 lvl=dbug msg="Shifting root filesystem \"/var/lib/lxd/containers/xenial/rootfs\" for \"xenial\"" 
t=2019-05-24T14:15:52+0200 lvl=dbug msg="Created DIR storage volume for container \"xenial\" on storage pool \"default\"" 
t=2019-05-24T14:15:52+0200 lvl=dbug msg="Success for task operation: 3edb1db8-639c-42e2-bd6e-0bfaa472b59a" 
t=2019-05-24T14:15:52+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/containers/xenial
t=2019-05-24T14:15:52+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:52+0200 lvl=dbug msg="Disconnected event listener: 0b03ccef-3cb1-464a-ba76-6a6d1b2026a6" 
t=2019-05-24T14:15:52+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:15:56+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:15:56+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:01+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:01+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:03+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:03+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/containers/xenial
t=2019-05-24T14:16:03+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/events?project="
t=2019-05-24T14:16:03+0200 lvl=dbug msg="New event listener: fbd9d795-a9cb-45f2-aec8-a0adf69990cf" 
t=2019-05-24T14:16:03+0200 lvl=dbug msg=handling ip=@ method=PUT url=/1.0/containers/xenial
t=2019-05-24T14:16:03+0200 lvl=dbug msg="New task operation: 745a9b8c-7b11-4fbf-ab27-b9e36040e479" 
t=2019-05-24T14:16:03+0200 lvl=dbug msg="Started task operation: 745a9b8c-7b11-4fbf-ab27-b9e36040e479" 
t=2019-05-24T14:16:03+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/operations/745a9b8c-7b11-4fbf-ab27-b9e36040e479
t=2019-05-24T14:16:03+0200 lvl=dbug msg="Success for task operation: 745a9b8c-7b11-4fbf-ab27-b9e36040e479" 
t=2019-05-24T14:16:05+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:05+0200 lvl=dbug msg="Disconnected event listener: fbd9d795-a9cb-45f2-aec8-a0adf69990cf" 
t=2019-05-24T14:16:05+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:09+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:09+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:13+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:13+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:17+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:17+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:22+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:22+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:26+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:26+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:27+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:27+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/containers/xenial
t=2019-05-24T14:16:27+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/events?project="
t=2019-05-24T14:16:27+0200 lvl=dbug msg="New event listener: 77a7b8fa-3148-432f-ab32-97e79f95abf0" 
t=2019-05-24T14:16:27+0200 lvl=dbug msg=handling ip=@ method=PUT url=/1.0/containers/xenial/state
t=2019-05-24T14:16:27+0200 lvl=dbug msg="New task operation: 1fd3a98d-e9a6-4ba1-b5b5-c086ee4ccf10" 
t=2019-05-24T14:16:27+0200 lvl=dbug msg="Started task operation: 1fd3a98d-e9a6-4ba1-b5b5-c086ee4ccf10" 
t=2019-05-24T14:16:27+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0/operations/1fd3a98d-e9a6-4ba1-b5b5-c086ee4ccf10
t=2019-05-24T14:16:27+0200 lvl=info msg="Starting container" action=start created=2019-05-24T14:15:45+0200 ephemeral=false name=xenial project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T14:16:27+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:27+0200 lvl=dbug msg=handling ip=@ method=GET url=/internal/containers/439/onstart
t=2019-05-24T14:16:27+0200 lvl=dbug msg="Scheduler: container xenial started: re-balancing" 
t=2019-05-24T14:16:27+0200 lvl=info msg="Started container" action=start created=2019-05-24T14:15:45+0200 ephemeral=false name=xenial project=default stateful=false used=1970-01-01T01:00:00+0100
t=2019-05-24T14:16:27+0200 lvl=dbug msg="Success for task operation: 1fd3a98d-e9a6-4ba1-b5b5-c086ee4ccf10" 
t=2019-05-24T14:16:30+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:30+0200 lvl=dbug msg="Disconnected event listener: 77a7b8fa-3148-432f-ab32-97e79f95abf0" 
t=2019-05-24T14:16:30+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
t=2019-05-24T14:16:34+0200 lvl=dbug msg=handling ip=@ method=GET url=/1.0
t=2019-05-24T14:16:34+0200 lvl=dbug msg=handling ip=@ method=GET url="/1.0/containers?recursion=2"
```

**Comment by stgraber** [comment]
That's normal and not a LXD issue.

What's happening here is that your mount is correctly processed by LXD/liblxc and mounted on your path under /run during setup. Then init in your container is spawned and mounts a new tmpfs on top of /run, hiding whatever you already mounted there.

There's nothing LXD can do about such behavior. You may be able to tweak your container's fstab or init system to not do that, or have LXD mount it somewhere else and then have an init script of fstab entry in the container move it to the correct spot during boot.

The same is true of /tmp, though /tmp is even worse as init systems actively wipe it during boot.
